src/Point.ts

```diff
@@ -1 +1,36 @@
-export class Point {}
+export class Point {
+  x: number;
+  y: number;
+
+  constructor();
+  constructor(x: number, y: number);
+  constructor(x?: number, y?: number) {
+    this.x = x ?? 0;
+    this.y = y ?? 0;
+  }
+
+  distance(): number;
+  distance(point: Point): number;
+  distance(x: number, y: number): number;
+  distance(x?: number | Point, y?: number): number {
+    let distance: number;
```

Comment:
Good job!

Just a small comment here. You could've written this structure without an additional variable and else statements:
```
        if (x instanceof Point) {
            return this.calculateDistance(x.x, x.y);
        }
        if (x !== undefined && y !== undefined) {
            return this.calculateDistance(x, y);
        }
        return this.calculateDistance(0, 0);
```
That way the code looks more concise and easy to read IMHO. It's more of the code style preference, and opinionated, so no need to edit this, just FYI.